setup.py

```diff
@@ -138,6 +140,7 @@ def run(self):
         "hdfs": hdfs,
         "webhdfs": webhdfs,
         "webdav": webdav,
+        "terraform": terraform,
```

Comment:
```suggestion
        "machine": terraform,
```

Reply:
:100: Or whatever name we choose for cloud execution capabilities.

Reply:
> Both `ssh` and `kubernetes` seem like odd values to provide for the `cloud` argument IMO. I guess we are copying that from the terraform-provider-iterative syntax. It seems more natural to call it something more generic like `provider` since that's the equivalent in terraform. Not a blocker, maybe it's just me 🤷 .

Yeah, it's `cloud` because that's the name of the TPI option. Basically I think the `.dvc/config` options should 1:1 map with the TPI ones. It doesn't really make sense for us to say "dvc config should have `provider = aws` but if you are using TPI yourself you need to use `cloud = aws`"

Reply:
I named the extra `terraform` for now because the `dvc machine` feature itself is supposed to be independent of whatever backend is actually used to provision machines (in this case, the only supported backend right now would be `terraform`). So in this case to be able to specifically use `dvc machine` with the terraform-based backend, the install extra is `terraform`.

Hypothetically, if someone else wanted to write a different backend (that wrapped something like docker-machine for provisioning instead of terraform) you could do that, and then install `dvc[docker-machine]` to use it instead of `dvc[terraform]`.

The more realistic example would be that eventually we may support arbitrary ssh machines in general (without any provisioning backend), in which case you should just be able to use `dvc machine` out of the box without installing any extras at all.

Reply:
yes fully agree, however... do we realistically think we may support other `machine` backends? As-is, TPI is a lot of work. For that reason I'm not convinced it's exactly the same as `dvc[remote]`.